File: 100-Days-Series/Day 45/PalindromeString.cpp
#include<iostream>
#include<string>
using namespace std;
// Fucntion to check if string is a Palindrome or not
bool getCheckPalindrome(string str){
    int start=0;
    int end=str.length()-1;
    while(start<end){
        if(str[start]!=str[end])
            return false;
        start++;
        end--;
    }
    return 1;

}
// Let's Start here
int main(){
    string str;
    cout<<"\nEnter a String : ";
    cin>>str;
    if(getCheckPalindrome(str))
        cout<<"\nThe String "<<str<<" is a Palindrome!"<<endl;
    else
        cout<<"\nThis String "<<str<<" is not a Palidrome!"<<endl;

    return 0;
}

File: 100-Days-Series/Day 45/PalindromeString.py

# Function to check if string is palindrome or not
def getCheckPlindrome(str):
    temp=""
    for i in range(len(str)-1,-1,-1):
        temp+=str[i]
    return temp==str

# Let's Start here
str=input('Enter a string : ')
if getCheckPlindrome(str):
    print("String is a Palindrome")
else:
    print("String is not a Palindrome")